src/data/nat/choose/bounds.lean

```diff
@@ -0,0 +1,44 @@
+/-
+Copyright (c) 2021 Yaël Dillies. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Yaël Dillies, Eric Rodriguez
+-/
+
+import data.nat.choose.basic
+import data.nat.cast
+import algebra.group_power.lemmas
+
+/-!
+# Inequalities for binomial coefficients
+
+This file proves exponential bounds on binomial coefficients. We might want to add here the
+bounds `n^r/r^r ≤ n.choose r ≤ e^r n^r/r^r` in the future.
+
+## Main declarations
+
+* `choose_le_pow`: `n.choose r ≤ n^r / r!`
+* `pow_le_choose`: `(n + 1 - r)^r / r! ≤ n.choose r`. Beware of the fishy ℕ-subtraction.
+-/
+
+open_locale nat
+
+variables {α : Type*} [linear_ordered_field α] [nontrivial α]
```

Comment:
Are you sure you need `nontrivial α` here? Otherwise, all the inequalities should be true, right?

Reply:
Every field is nontrivial, isn't it? `0 \ne 1`

Reply:
Oh, some moving around of caused this.